sycl/include/sycl/ext/intel/esimd/common.hpp

```diff
@@ -106,6 +66,16 @@ enum class rgba_channel : uint8_t { R, G, B, A };
 using SurfaceIndex = unsigned int;
 
 namespace detail {
+
+/// Check if a given 32 bit positive integer is a power of 2 at compile time.
+ESIMD_INLINE constexpr bool isPowerOf2(unsigned int n) {
+  return (n & (n - 1)) == 0;
+}
+
+ESIMD_INLINE constexpr bool isPowerOf2(unsigned int n, unsigned int limit) {
```

Comment:
Minor comment: May need 1 line comment just for consistency with isPowerOf2 at L70, which has got such a comment.